codeframe/cli.py

```
@@ -163,6 +169,94 @@ def agents(
+
+        browser_thread = threading.Thread(target=open_in_browser, daemon=True)
+        browser_thread.start()
+
+    # Start server (blocking call)
+    try:
+        subprocess.run(cmd, check=True, capture_output=True, text=True)
+    except KeyboardInterrupt:
+        console.print("\n✓ Server stopped")
+    except FileNotFoundError:
+        console.print("[red]Error:[/red] uvicorn not found. Install with: pip install uvicorn")
+        raise typer.Exit(1)
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)